### PR TITLE
Wrap getCredentials() in a doPrivileged() block

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This commit fixes an issue that was missed in #22534.
`AWSCredentialsProvider.getCredentials()` appears to potentially open a
socket connect. This operation needed to be wrapped in `doPrivileged()`.

This should fix issue #23271.

### CODE REVIEW

**Comment by tbrooks8** [comment]
I tested this manually with the scenario provided by @dadoonet in #23271.

**Comment by rjernst** [comment]
Hrm, that is not the only place `getCredentials()` will be called. It would also be called by the s3 client.  I think instead, the doPriv should be inside a wrapper credentials provider, setup to wrap the InstanceProfileCredentialsProvider created in buildCredentials?

**Comment by tbrooks8** [comment]
I adjusted the PR to reflect @rjernst's review.

File: plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/InternalAwsS3Service.java [hunk 1]
```diff
@@ -156,7 +158,17 @@ public static AWSCredentialsProvider buildCredentials(Logger logger, Deprecation
             }
         }
 
-        return credentials;
+        return new AWSCredentialsProvider() {
+            @Override
+            public AWSCredentials getCredentials() {
+                return SocketAccess.doPrivileged(credentials::getCredentials);
```
Comment (rjernst) [inline plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/InternalAwsS3Service.java]: We only need this for the instance profile credentials. The rest are read and created statically, so definitely do not require socket access.

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**